sec2-architecture.tex

```diff
@@ -155,91 +188,137 @@ \subsubsection{LSST-specific Objects}\label{lsst-specific-objects}
+is provided in Section~\ref{api-aspect} below.
 
-Within the JupyterLab Aspect, and internal to the Web APIs Aspect, the Data
+Within the Notebook Aspect, and internal to the Web API Aspect, the Data
 Butler client interface may be used to retrieve Python objects, and a
-"native" Data Backbone API may be used for high-performance retrieval of
-datasets.  The Portal Aspect uses the Web APIs Aspect exclusively for data
-access; it does not use the Data Butler nor the "native" Data Backbone API.
+``native'' Data Backbone API may be used for high-performance retrieval of
+datasets.
+The Portal Aspect primarily uses the Web API Aspect for data access, 
+though the Data Butler and ``native'' Data Backbone API may be used in Python
+plugins that provide support for access to, and visualization of, LSST-specific
+complex datatypes.
 
 \subsubsection{User Catalog Data Support}\label{user-catalog-data-support}
 
 Users may create their own catalogs within the LSP, a capability that is
-patterned after SDSS SkyServer's "MyDB".  Catalogs may be derived from queries
+patterned after SDSS SkyServer's ``MyDB''.  
+Catalogs may be derived from queries
 to the LSST catalogs or loaded from computations or external data sources.  As
 mentioned above in section~\ref{database-technologies}, large distributed
-catalogs meant for joining with the LSST data products must be identified so
-that they can be stored appropriately in the Qserv database.
+catalogs meant for joining with the LSST data products must be identified by
+the user so that they can be stored appropriately in the Qserv database.
 
 All three Aspects will provide mechanisms for creating user catalogs; the
-Portal's mechanism will be implemented via the Web APIs, as usual.  User
-catalogs may be accessed in the same ways that LSST catalog data products are
-accessed.
+Portal's mechanism will be implemented via the Web APIs, as usual.
+User catalogs may be accessed in the same ways that LSST catalog data products
+are accessed, and are visible in every Aspect.
 
 User catalogs are backed up for disaster recovery purposes, but they are not
 fully protected against user error.
 
-\subsubsection{User Workspace Storage}\label{user-workspace-storage}
-
-Users may create their own images and files within the LSP.  These all live
-within a User Workspace that is common across all three Aspects.  All three
-Aspects will provide mechanisms for creating and retrieving user files.  In
-the JupyterLab Aspect, this is via a mounted filesystem.  In the Web APIs
-Aspect, this is via VOSpace or WebDAV.
+\subsubsection{User File Workspace Storage}\label{user-workspace-storage}
+
+Users may create their own images and other files within the LSP.
+These all live within a User Workspace that is common across all three Aspects.
+All three Aspects will provide mechanisms for creating and retrieving user
+files.
+In the Notebook Aspect, this is via a mounted filesystem.  
+In the Web API Aspect, this is via VOSpace or WebDAV.
```

Comment:
I think it was VOSpace and maybe WebDAV in addition to that. Also, does this document mention shared workspaces for collaboration? I'd also like to be able to make a VOSpace area as readonly and issue a DOI for it (CADC allow this).

Reply:
1. You are correct about what we say about VOSpace and WebDAV.  This was clearer in Section 5.  I'm fixing that here as well.
1. The concept of sharing workspaces was mentioned in a few places but was not given the appropriate prominence in this document compared to its importance in the design.  I'll fix that.
1. I suggest that we take the question of "freezing" data in the database and/or file workspaces and defining a DOI to a separate discussion and perhaps RFC.  There are too many issues for that to be able to be settled by a minor edit here.